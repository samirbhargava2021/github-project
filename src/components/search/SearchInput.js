import React, { useState } from "react";
import Button from "../../commonComponents/button/Button";
import "./SearchInput.scss";

const SearchInput = ({ onSearch }) => {
  const [searchText, setSearchText] = useState("");
  return (
    <div className="search-container">
      <input
        type="search"
        name="searchText"
        value={searchText}
        className="search-container__search-input"
        aria-label="search"
        placeholder="Enter Repository name.."
        onChange={(event) => setSearchText(event.target.value)}
      />
      <Button 
        events={() => onSearch(searchText)}
        data={{
          theme:"light",
          text:"Search"
        }}
        options={{
            buttonSize:"small"
        }}
      />
    </div>
  );
};

export default SearchInput;
