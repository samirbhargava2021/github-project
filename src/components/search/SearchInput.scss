.search-container {
  margin: 20px;
  display: flex;
  justify-content: center;
  &__search-input {
    width: 80%;
    padding: 16px;
    border: 1px solid #d7dcdf;
  }
  &__search-btn {
    width: 180px;
    border: 1px solid #fce4ec;
    padding: 12px 6px;
    background: #ea1e62;
    color: #fff;
    cursor: pointer;
    font-weight: 600;
    border-radius: 4px;
    margin-left: 12px;
  }
}
