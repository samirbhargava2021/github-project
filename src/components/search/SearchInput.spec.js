import { shallow } from "enzyme";
import { render, fireEvent  } from "@testing-library/react";
import SearchInput from "./SearchInput";

describe("SeachInput Component", () => {
    const setup = () => {
        const utils = render(<SearchInput onSearch='react'/>)
        const input = utils.getByLabelText('search')
        return {
          input,
          ...utils,
        }
    }
  it("should render Search Input Component", () => {
    const component = shallow(<SearchInput />);
    expect(component.getElements()).toMatchSnapshot();
  });
  it("should have className search-container to load the searchInput with proper styling", () => {
    const { container } = render(<SearchInput />);
    expect(container.firstChild.classList.contains('search-container')).toBe(true)
  });
  it('It should check if the input value changes', () => {
    const { input } = setup();
    fireEvent.change(input, { target: { value: 'angular' } })
    expect(input.value).toBe('angular');
  });
});
