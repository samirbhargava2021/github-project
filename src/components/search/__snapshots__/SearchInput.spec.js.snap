// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeachInput Component should render Search Input Component 1`] = `
Array [
  <div
    className="search-container"
  >
    <input
      aria-label="search"
      className="search-container__search-input"
      name="searchText"
      onChange={[Function]}
      placeholder="Enter Repository name.."
      type="search"
      value=""
    />
    <Button
      data={
        Object {
          "text": "Search",
          "theme": "light",
        }
      }
      events={[Function]}
      options={
        Object {
          "buttonSize": "small",
        }
      }
    />
  </div>,
]
`;
